components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -746,3 +745,81 @@ interface IAPSubscription {
   GetPurchaseTokenOrderId() => (string token, string order_id);
   SetLinkStatus(int32 status);
 };
+
+// Interfaces for communication between the untrusted content frame and both
+// the Browser and the parent trusted UI frame.
+
+// Trusted WebUI-side handler for messages from the untrusted child frame
+interface ParentUIFrame {
+  // <iframe> cannot be sized to the natural height of its content, so the child
+  // must let the parent know whenever the body height changes so that the
+  // parent can manually set the same height on the <iframe> to avoid scrolling.
+  ChildHeightChanged(uint32 height);
+
+  // This is sent to the parent trusted webui rather than straight to the
+  // browser to allow for further rating feedback to be attached. The parent
+  // frame deals with it instead of the child frame, due to UI space as well as
+  // sensitive user information being captured.
+  RateMessage(string turn_uuid, bool is_liked);
+
+  // This is sent to the parent trusted webui to allow displaying the
+  // open external link modal outside of the untrusted iframe. This is
+  // to prevent the modal backdrop from being clipped within the iframe
+  // and fixes centering the modal in the page view. The parent frame
+  // then deals with whether to display the modal or to continue opening
+  // the link.
+  UserRequestedOpenGeneratedUrl(url.mojom.Url url);
+
+  // Notify the parent that the child has received a drag enter. We don't have
+  // a corresponding DragEnd because the parent should intercept the drag event
+  // with an overlay.
+  DragStart();
+
+  // This is sent to the parent trusted webui to allow settings the iframes
+  // wrapper position to relative while the Regenerate Answer Menu is open.
+  // Otherwise the menu can sometimes be overlapped by the Suggested question
+  // buttons.
+  RegenerateAnswerMenuIsOpen(bool is_open);
+
+  // Show the smart mode creation dialog with the provided prompt text
+  ShowSmartModeDialog(string prompt);
+};
+
+// UI-side handler for callbacks from UntrustedUIHandler
+interface UntrustedUI {
+  // Called when memories have changed (added, updated, or deleted)
+  OnMemoriesChanged(array<string> memories);
+};
+
+// Browser-side handler for untrusted frame that handles rendering of
+// conversation entries.
+interface UntrustedUIHandler {
+  // Bind the conversation handler
+  BindConversationHandler(
+    string conversation_id,
+    pending_receiver<UntrustedConversationHandler> conversation_handler
+  );
```

Comment:
This seems like a sensible way to get around not being able to iterate the subframes on ios, and doesn't really affect desktop either :smile: